usr/sbin/p11sak/p11sak.c

```diff
@@ -103,23 +103,14 @@ static CK_RV get_pin(char **pin, size_t *pinlen)
     printf("\n");
     fflush(stdout);
 
-    /* Allocate  PIN.
-     * Note: nread includes carriage return.
-     * Replace with terminating NULL.
-     */
-    *pin = (char*) malloc(nread);
-    if (*pin == NULL) {
-        rc = -ENOMEM;
-        goto done;
-    }
-
     /* strip the carriage return since not part of pin. */
     user_input[nread - 1] = '\0';
```

Comment:
Could we make this conditional while your are at it?  `getline` also returns on EOF-terminated lines.  Only if the line is empty and EOF is received you get `-1`.

Reply:
Will do.